### PR TITLE
feat: Add configurable user_session_timeout

### PR DESCRIPTION
**Pull request by wholesomewilson**
Addresses #1031

This adds `user_session_timeout` variable to config.toml to make user session timeout configurable.

### CODE REVIEW

**Comment by Physium** [comment]
Any updates if this can be merged?

File: backend/chainlit/config.py [hunk 1]
```diff
@@ -51,6 +51,9 @@
 # Duration (in seconds) during which the session is saved when the connection is lost
 session_timeout = 3600
 
+# Duration (in seconds) of the user session expiry
+user_session_timeout = 1296000
```
Comment (dokterbob) [inline backend/chainlit/config.py:55]: I would suggest adding that it's 15 days to the comment here.

File: backend/chainlit/config.py [hunk 2]
```diff
@@ -297,6 +300,8 @@ class ProjectSettings(DataClassJsonMixin):
     # Path to the local chat db
     # Duration (in seconds) during which the session is saved when the connection is lost
     session_timeout: int = 3600
+    # Duration (in seconds) of the user session expiry
+    user_session_timeout: int = 60 * 60 * 24 * 15
```
Comment (dokterbob) [inline backend/chainlit/config.py:304]: I'd suggest adding comment on 15 days here.

**Review by dokterbob — CHANGES_REQUESTED** [review]
The feature is well explained and clearly scoped and code looks clean -- but outdated. Could you please rebase and add requested comments?

If tests pass I think we should be good to go.

File: backend/chainlit/config.py (hunk 1 above)
Comment (aniruddha-adhikary) [reply backend/chainlit/config.py]: We just added a comment there to suggest `15 days`.

File: backend/chainlit/config.py (hunk 2 above)
Comment (aniruddha-adhikary) [reply backend/chainlit/config.py]: Comment added as well!

**Comment by aniruddha-adhikary** [comment]
Hi @dokterbob, we have made the changes requested.

**Pull request merged**